src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/transcripts-list-item-info/TranscriptsListItemInfo.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import { getCommaSeparatedNumber } from 'src/shared/helpers/formatters/numberFormatter';
+import {
+  getFeatureCoordinates,
+  getRegionName
+} from 'src/content/app/entity-viewer/shared/helpers/entity-helpers';
+
+import { Transcript } from 'src/content/app/entity-viewer/types/transcript';
+
+import transcriptsListStyles from '../DefaultTranscriptsList.scss';
+import styles from './TranscriptsListItemInfo.scss';
+
+type ItemInfoProps = {
+  transcript: Transcript;
+};
+
+const ItemInfo = (props: ItemInfoProps) => {
+  const { transcript } = props;
+
+  const getTranscriptLocation = () => {
+    const { start, end } = getFeatureCoordinates(transcript);
+    const startStr = getCommaSeparatedNumber(start);
+    const endStr = getCommaSeparatedNumber(end);
+    const regionName = getRegionName(transcript);
+    return `${regionName}:${startStr}-${endStr}`;
+  };
+
+  const getRNALength = () => {
+    const rnaLength = transcript.exons.reduce((length, exon) => {
+      const { start, end } = getFeatureCoordinates(exon);
+      return length + (end - start + 1);
+    }, 0);
+
+    return getCommaSeparatedNumber(rnaLength);
+  };
+
+  const getFirstAndLastCodingExonIndexes = () => {
+    const { exons, cds } = transcript;
+    let firstCodingExonIndex = 0;
+    let lastCodingExonIndex = exons.length - 1;
+
+    if (cds) {
+      for (let index = 0; index < exons.length; index += 1) {
+        const exonLocation = getFeatureCoordinates(exons[index]);
+
+        if (exonLocation.start <= cds.start && exonLocation.end >= cds.start) {
+          firstCodingExonIndex = index;
+          break;
+        }
+      }
+
+      for (let index = exons.length - 1; index > -1; index -= 1) {
+        const exonLocation = getFeatureCoordinates(exons[index]);
+
+        if (exonLocation.start <= cds.end && exonLocation.end >= cds.end) {
+          lastCodingExonIndex = index;
+          break;
+        }
+      }
```

Comment:
Aren't these two for-loops better expressed as Array.prototype.find?

Reply:
✔️